src/NEventStore/Persistence/Sql/SqlPersistenceEngine.cs

```diff
@@ -92,13 +96,39 @@ public virtual void Initialize()
             ExecuteCommand(statement => statement.ExecuteWithoutExceptions(_dialect.InitializeStorage));
         }
 
-        public virtual IEnumerable<ICommit> GetFrom(string bucketId, string streamId, int minRevision, int maxRevision)
+        public virtual IEnumerable<ICommit> GetFromSnapshot(ISnapshot snapshot, int maxRevision)
         {
+            var streamId = snapshot.StreamId;
+            var minRevision = snapshot.StreamRevision;
+
             Logger.Debug(Messages.GettingAllCommitsBetween, streamId, minRevision, maxRevision);
             streamId = _streamIdHasher.GetHash(streamId);
             return ExecuteQuery(query =>
+            {
+                string statement = _dialect.GetCommitsFromStartingRevision;
+                query.AddParameter(_dialect.BucketId, snapshot.BucketId, DbType.AnsiString);
+                query.AddParameter(_dialect.StreamId, streamId, DbType.AnsiString);
+                query.AddParameter(_dialect.StreamRevision, minRevision);
+                query.AddParameter(_dialect.MaxStreamRevision, maxRevision);
+                query.AddParameter(_dialect.CommitSequence, 0);
+                return query
+                    .ExecutePagedQuery(statement, _dialect.NextPageDelegate)
+                    .Select(x => x.GetCommit(_serializer, _dialect));
+            });
+        }
+
+        public virtual IEnumerable<ICommit> GetFrom(string bucketId, string streamId, int minRevision, int maxRevision)
+        {
+            Logger.Debug(Messages.GettingAllCommitsBetween, streamId, minRevision, maxRevision);
+            streamId = _streamIdHasher.GetHash(streamId);
+
+            var commits = new List<ICommit>();
+            if(_archivingConnection != null)
```

Comment:
I don't know if this has any impact on paging, not sure how the `_dialect.NextPageDelegate` thing work. Querying from multiple sources is always a pain when it comes down to paging for me.

Reply:
Yeh I thought the same but luckily it returns all the Commits in this method but it grabs them a page at a time from the DB internally.

Reply:
What I meant was that if you are adding a page here and adding another page after this `if` block, wouldn't it make the result double the page size?

Reply:
No because it returns all the results, not a page. 

`return query.ExecutePagedQuery(statement, (q, r) => { })
                    .Select(x => x.GetCommit(_serializer, _dialect));`

This will get all the results from the DB but internally it will retrieve a page at a time until it has them all.